spark-submit.sh

```diff
@@ -0,0 +1,2 @@
+export JAVA_HOME=/usr/java/jdk1.8.0_162 && ./bin/spark2-submit --class com.gd.twitteranalytics.StreamingTweetsJob \
```

Comment:
the script should be made environment independent / generic.

> /usr/java/jdk1.8.0_162

what if  I as a team member don't have java installed at this particular path?

> /home/ssachdeva/twitter-analytics-assembly-0.1.jar

why do I as a team member have to have "/home/**ssachdeva**" on my local environment to be able to run the script?

Reply:
Other alternate is to update the env.sh file. I have no idea as a team member I have rights to update the file or if it won't affect any other's team member job..

Do we have a specific path where jars are kept to which I have access ?

Reply:
> Do we have a specific path where jars are kept to which I have access ?
this is not quite right approach to follow...

In general, depending on a use-case, dynamic / environment dependent variables should be supplied as function agruments, external config, system/env variables file or cmd arguments. 

How would you implement it by following the design principle?

Reply:
Sorry, I am not getting this. Do we need to provide all dynamic variables via config file ? We are providing it as cmd arguments right now.. Still I need to know the path where jar should be kept. I check on the cluster and there are multiple JDKs..If I don't set JAVA_HOME, I get major minor error due to JDK mismatch.
Do we need to create a bash script ?

Reply:
Or maybe you simply mean to put dynamic values in conf file instead of hardcoding it in script file !